### PR TITLE
Added an example for data lineage with PostgreSQL queries

### PR DESCRIPTION
**Pull request by shanewholloway-usds**
Second transformation of Jim's tech lineage demonstration. This example focuses on data linage queries, and uses fake data generation behind the scenes.

## Changes
- Ported SQL code from `docs/proposals/0003-lineage/` into example oriented material.
- Split sequential demo into independent bite sized files
- Made `psql` commands work inside local docker container
- Minor formatting improvements for `examples/upload-fake-data/example.sh`

## Checklist
- [x] Tested locally?

### CODE REVIEW

File: prime-router/examples/data-lineage/lineage-report-stored-procedures.sql
Comment (jimduff-usds) [inline prime-router/examples/data-lineage/lineage-report-stored-procedures.sql:1]: These functions will now be permanently in the database, so they don't need to be manually loaded any more.

File: prime-router/examples/data-lineage/example-setup.sh
Comment (jimduff-usds) [inline prime-router/examples/data-lineage/example-setup.sh:9]: This step won't be needed any more - the stored process are going into the db permanently, once jim/lineage3 is merged.

**Review by jimduff-usds — CHANGES_REQUESTED** [review]
This looks great - the extraction of the report_id from the json is a nice improvement!

I have just one change request:  once the jim/lineage3 branch is merged, hopefully today, the base recursive lineage queries will be in the database, so you won't need to load them here.  (And if this is run against production ever, we wouldn't want to be loading code anyway).    So can you wait for that merge, then tweak the script here?

Also a suggestion:   You'll see in that branch that I added one more example inside "descendants.sql", to find all the "cousins" of a report.   This is a use case that @jlusds  has requested.   "I want to know all the other health departments that also got this same data".  So it would be good to add that to our example set here.

File: prime-router/examples/data-lineage/example-setup.sh
Comment (shanewholloway-usds) [reply prime-router/examples/data-lineage/example-setup.sh:9]: Merged in from master. Confirmed the stored procedures are in the clean DB.

File: prime-router/examples/data-lineage/lineage-report-stored-procedures.sql
Comment (shanewholloway-usds) [reply prime-router/examples/data-lineage/lineage-report-stored-procedures.sql:1]: Even better! Removed.

**Comment by shanewholloway-usds** [comment]
Merged from master and integrated feedback from @jimduff-usds

**Comment by jimduff-usds** [comment]
@shanewholloway , that branch I mentioned has now been merged into master.

**Review by MauriceReeves-usds — APPROVED** [review]
Looks great. Documentation is very thorough. Great work.

**Review by jimduff-usds — APPROVED** [review]

**Pull request merged**
